hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/ipc/CallerContext.java

```diff
@@ -108,24 +117,86 @@ public String toString() {
   }
 
   /** The caller context builder. */
-  public static final class Builder {
-    private final String context;
+  public static final class Builder implements Configurable {
```

Comment:
Great suggestions!

>Add a new method public append(String key, String value) which calls
> sb.append(key).append(separator).append(value)

For example, maybe context contains multi items and it is "mr_xxxx,ip:a.b.c.d", here item separator is ',' and key/value separator is ':', the separator we define is item separator. should we add method public append(String key, String value, separator)?

Reply:
It's straightforward to always use : to split key:value and do not allow user to change that. We can allow user to specify the item separator from Configuration but it can not be `=`, `\n`, `\t` or `:` etc

So we do not need the append(key, value, separator) method, since the `append(key, value)` simply appends `$separator,key:value` where separator is final from Configuration (or default).

Does this make sense?

Reply:
It's ok. Get it.